src/gwt/src/org/rstudio/studio/client/workbench/addins/Addins.java

```diff
@@ -143,10 +146,31 @@ public void execute()
                   {
                      String code = addin.getPackage() + ":::" + 
                                    addin.getBinding() + "()";
-                     events_.fireEvent(new SendToConsoleEvent(code, 
-                                                              true, 
-                                                              false, 
-                                                              false));
+                     
+                     if (BrowseCap.isMacintoshDesktopMojave())
+                     {
+                        server_.prepareForAddin(new ServerRequestCallback<Void>()
+                        {
+                           @Override
+                           public void onResponseReceived(Void response)
+                           {
+                              SendToConsoleEvent event = new SendToConsoleEvent(code, true, false, false);
```

Comment:
If an error occurs in the add-in such that the Shiny response never happens, then our response filter will be left dangling. I can't think offhand whether there's a good way to prevent this, though -- what do you think?

Reply:
One possibility would be for us to clear the filter on the next ConsolePromptEvent or something similar -- that might be more robust overall (and would allow us to properly handle the case where an addin might attempt a reload, however rare that might be)